### PR TITLE
Prioritize hgln_obs over crln_obs when extracting observer information from FITS headers

### PR DESCRIPTION
**Pull request by svank**
Fix as proposed in #7184

Thanks @ayshih for confirming this should be right!

### CODE REVIEW

File: sunpy/coordinates/wcs_utils.py [hunk 1]
```diff
@@ -129,6 +129,7 @@ def solar_wcs_frame_mapping(wcs):
                              attrs[1] * u.deg,
                              attrs[2] * u.m,
                              **kwargs)
+            break
```
Comment (dstansby) [inline sunpy/coordinates/wcs_utils.py:132]: Could you add a quick comment above where the `required_attrs` dictionary is defined that the order of entries matters there?

**Review by dstansby — CHANGES_REQUESTED** [review]
Looks 👍 , just one request for a comment, and needs a bugifx changelog entry adding.

**Comment by svank** [comment]
@dstansby Thanks! I've made these changes.

File: sunpy/coordinates/tests/test_wcs_utils.py [hunk 2]
```diff
@@ -399,3 +399,51 @@ def test_obsgeo_invalid(obsgeo):
...
     with pytest.raises(ValueError):
         obsgeo_to_frame(obsgeo, None)
+
+
+def test_observer_hgln_crln_priority():
+    """
+    When extracting the observer information from a FITS header, ensure
+    Stonyhurst (HG) coordinates are preferred over Carrington (CR) if present.
+    Going through `Map` and `WCS` seem to follow (at least slightly) different
+    code paths, so test them both.
```
Comment (ayshih) [inline sunpy/coordinates/tests/test_wcs_utils.py:409]:
```suggestion
    `Map` creates a custom FITS header with a sanitized observer location, so
    we need to also test a directly instantiated `WCS` object.
```

**Review by ayshih — CHANGES_REQUESTED** [review]
Some text edits, otherwise looks good to me!

If I were to be super-picky, the part of the unit test that tests the prioritization in the `Map` implementation should really live under `sunpy.map.tests` and not `sunpy.coordinates.tests`, because the relevant code – how `GenericMap` prioritizes and sanitizes the observer location – all lives in `sunpy/map/mapbase.py`.

**Comment by Cadair** [comment]
>  the part of the unit test that tests the prioritization in the Map implementation should really live under sunpy.map.tests

Given my recent foray into seeing if it's possible to run tests for subpackages without optional dependencies installed, ideally we would not have a direct reliance on map in the coordinates tests.

**Comment by svank** [comment]
What's code review for, if not being picky? I moved the `Map` part of the test to the `map` module---there is still a pre-existing `Map` dependence in `coordinates/tests/test_wcs_utils.py`, but now I avoid further entrenching that.

Text edits are also implemented.

**Review by ayshih — APPROVED** [review]
Looks good to me!

**Comment by ayshih** [comment]
Thanks, @svank!

**Pull request merged**